services/motion/builtin/wrapperFrame.go

```diff
@@ -0,0 +1,144 @@
+func (wf *wrapperFrame) Transform(inputs []referenceframe.Input) (spatialmath.Pose, error) {
+	if len(inputs) != len(wf.DoF()) {
+		return nil, referenceframe.NewIncorrectInputLengthError(len(inputs), len(wf.DoF()))
+	}
+	executionFramePose, err := wf.executionFrame.Transform(inputs[:len(wf.executionFrame.DoF())])
+	if err != nil {
+		return nil, err
+	}
+	localizationFramePose, err := wf.localizationFrame.Transform(inputs[len(wf.executionFrame.DoF()):])
+	if err != nil {
+		return nil, err
+	}
+	return spatialmath.Compose(localizationFramePose, executionFramePose), nil
+}
+
+// Interpolate interpolates the given amount between the two sets of inputs.
+func (wf *wrapperFrame) Interpolate(from, to []referenceframe.Input, by float64) ([]referenceframe.Input, error) {
+	if len(from) != len(wf.DoF()) {
+		return nil, referenceframe.NewIncorrectInputLengthError(len(from), len(wf.DoF()))
+	}
+	if len(to) != len(wf.DoF()) {
+		return nil, referenceframe.NewIncorrectInputLengthError(len(to), len(wf.DoF()))
+	}
+	interp := make([]referenceframe.Input, 0, len(to))
+
+	// both from and to are lists with length = 11
+	// the first four values correspond to ptg(execution) frame values we want the interpolate
+	// the latter seven values correspond to pose(localization) frame values we want the interpolate
+
+	// executionFrame interpolation
+	executionFrameFromSubset := make([]referenceframe.Input, len(wf.executionFrame.DoF()))
+	executionFrameToSubset := to[:len(wf.executionFrame.DoF())]
+	interpSub, err := wf.executionFrame.Interpolate(executionFrameFromSubset, executionFrameToSubset, by)
+	if err != nil {
+		return nil, err
+	}
+	interp = append(interp, interpSub...)
+
+	// localizationFrame interpolation
+	// interpolating the localizationFrame is a special case
+	// the ToSubset of the localizationFrame does not matter since the executionFrame interpolations
+	// move us through a given segment and the localizationFrame input values are what we compose with
+	// the position ourselves in our absolute position in world.
+	localizationFrameSubset := to[len(wf.executionFrame.DoF()):]
+	interp = append(interp, localizationFrameSubset...)
+
+	return interp, nil
+}
+
+// Geometries returns an object representing the 3D space associated with the executionFrame's geometry.
+func (wf *wrapperFrame) Geometries(inputs []referenceframe.Input) (*referenceframe.GeometriesInFrame, error) {
+	if len(inputs) != len(wf.DoF()) {
+		return nil, referenceframe.NewIncorrectInputLengthError(len(inputs), len(wf.DoF()))
+	}
+	sfGeometries := []spatialmath.Geometry{}
+	gf, err := wf.executionFrame.Geometries(make([]referenceframe.Input, len(wf.executionFrame.DoF())))
+	if err != nil {
+		return nil, err
+	}
+	transformBy, err := wf.Transform(inputs)
+	if err != nil {
+		return nil, err
+	}
+	for _, g := range gf.Geometries() {
+		sfGeometries = append(sfGeometries, g.Transform(transformBy))
+	}
+	return referenceframe.NewGeometriesInFrame(referenceframe.World, sfGeometries), nil
```

Comment:
Shouldn't this be the same frame as whatever `gf` was in, rather than `World`?

Reply:
Or more like, in frame of `wf`?

Reply:
These frames may not always have `World` as a parent; if we were to introduce code in `motionplan` that constructed a rearranged framesystem (potentially useful), these would then return geometries in the wrong places.

The inputs place the geometries where they are relative to wherever the pose frame is rooted; since geometry poses are considered relative to the *root* of their parent frame, they will be correctly placed.

Reply:
>since geometry poses are considered relative to the root of their parent frame, they will be correctly placed.

ahh gotcha. I will update such that the parent here is `wf`